inference-engine/src/gna_plugin/backend/gna_limitations.cpp

```diff
@@ -182,22 +183,68 @@ bool AreLayersSupported(InferenceEngine::CNNNetwork& network, std::string& errMe
+                                               if (batch_size != 1 && info.isBatchSizeConstrained(layer->type)) {
                                                    errMessage = "topology with layer: " + layer->name + ", type: " + layer->type +
                                                                 ", and batch size(" + std::to_string(batch_size) + ") != 1 not supported";
                                                    check_result =  false;
                                                }
-                                               if (GNAPluginNS::LayerInfo(layer).isFullyConnected()) {
-                                                   size_t output_batch_size = GNAPluginNS::LayerInfo(layer).getOutputBatchSize();
+                                               if (info.isFullyConnected()) {
+                                                   size_t output_batch_size = info.getOutputBatchSize();
                                                    if (output_batch_size > 8) {
                                                         errMessage = "topology with layer: " + layer->name + ", type: " + layer->type +
                                                                      ", and batch size(" + std::to_string(output_batch_size) + ") not supported";
                                                         check_result =  false;
                                                    }
+                                               } else if (info.isConcat()) {
+                                                   auto concatLayer = info.as<InferenceEngine::ConcatLayer*>();
+                                                   IE_ASSERT(concatLayer);
+                                                   auto dims_size = concatLayer->insData[0].lock()->getDims().size();
+                                                   InferenceEngine::CNNLayerPtr prevLayer;
+
+                                                   auto isFusableWithConv = [](InferenceEngine::CNNLayerPtr ptr) {
+                                                       return LayerInfo(ptr).isFusableWithConv();
+                                                   };
+
+                                                   for (auto input_idx = 0; input_idx != concatLayer->insData.size(); input_idx++) {
+                                                       // Concatenation of a convolution in W dimension is supported
+                                                       // when the network layout is NCHW and there are no transposes around the convolution
+                                                       prevLayer = InferenceEngine::CNNNetPrevLayerSkipCertain(layer, input_idx, isFusableWithConv);
+                                                       if (prevLayer && LayerInfo(prevLayer).isConvolution() &&
+                                                           prevLayer->input()->getLayout() == InferenceEngine::Layout::NCHW)
```

Comment:
I think the whole verification of concat should be moved to a later stage (primitive creation), although it might be a bit less informative for the user.